make/e2e-ci.sh

```diff
@@ -16,4 +16,4 @@
 
 set -o errexit
 trap 'make kind-logs' EXIT
-make --no-print-directory e2e FLAKE_ATTEMPTS=2 K8S_VERSION="$(K8S_VERSION)"
+make --no-print-directory e2e FLAKE_ATTEMPTS=2 CI=true K8S_VERSION="$(K8S_VERSION)"
```

Comment:
`CI` should have been set by th GH worker/ prow worker already, right?
Is this change for local development?

Reply:
It is set by prow, yeah. I force it here for both local testing and because we explicitly call `make e2e-ci` in prow. It's just clearer IMO!

Reply:
Would you mind adding a small comment explain this? This is something that would be confusing to me when I want to figure out why we set the variable here.

Reply:
Makes sense, yeah, I'll do that!

Reply:
Done!